src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -0,0 +1,124 @@
+        raise CLIError(
+            'The input datetime is invalid. Correct format should be YYYY-MM-DDThh:mm:ssZ ')
+
+
+def validate_connection_string(namespace):
+    ''' Endpoint=https://example.azconfig.io;Id=xxxxx;Secret=xxxx'''
+    connection_string = namespace.connection_string
+    if connection_string:
+        if not is_valid_connection_string(connection_string):
+            raise CLIError('''The connection string is invalid. \
+Correct format should be Endpoint=https://example.azconfig.io;Id=xxxxx;Secret=xxxx ''')
+
+
+def validate_import_depth(namespace):
+    depth = namespace.depth
+    if depth is not None:
+        try:
+            depth = int(depth)
+            if depth < 1:
+                raise CLIError('Depth should be at least 1.')
+        except ValueError:
+            raise CLIError("Depth is not a number.")
+
+
+def validate_separator(namespace):
+    if namespace.separator is not None:
+        if namespace.format_ == "properties":
+            raise CLIError("Separator is not needed for properties file.")
+        valid_separators = ['.', ',', ';', '-', '_', '__', '/', ':']
+        if namespace.separator not in valid_separators:
+            raise CLIError(
+                "Unsupported separator, allowed values: '.', ',', ';', '-', '_', '__', '/', ':'.")
+
+
+def validate_import(namespace):
+    source = namespace.source
+    if source == 'file':
+        if namespace.path is None:
+            raise CLIError("Please specify the file path.")
+        if namespace.format_ is None:
+            raise CLIError("Please specify the file format.")
+    elif source == 'appconfig':
+        if (namespace.src_name is None) and (namespace.src_connection_string is None):
+            raise CLIError(
+                "Please specify either the AppConfig name or its connection string.")
+    elif source == 'appservice':
+        if namespace.appservice_account is None:
+            raise CLIError(
+                "Please specify the name of the AppService account.")
+
+
+def validate_export(namespace):
+    destination = namespace.destination
+    if destination == 'file':
+        if namespace.path is None:
+            raise CLIError("Please specify the file path.")
+        if namespace.format_ is None:
+            raise CLIError("Please specify the file format.")
+    elif destination == 'appconfig':
+        if (namespace.dest_name is None) and (namespace.dest_connection_string is None):
+            raise CLIError(
+                "Please specify either the AppConfig name or its connection string.")
+    elif destination == 'appservice':
+        if namespace.appservice_name is None:
+            raise CLIError("Please specify the name of the AppService.")
+        if namespace.appservice_resource_group is None:
+            raise CLIError(
+                "Please specify the name of the AppService's resource group.")
```

Comment:
appservice_resource_group directly violates the CLI command guidelines:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters

Reply:
Remove the check for resource group. Instead user can pass in name or just ID.